src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs

```diff
@@ -85,8 +102,21 @@ public async Task StartAsync()
         /// <returns><c>true</c> if delay between next time we should ask for blocks is required; <c>false</c> otherwise.</returns>
         protected async Task<bool> SyncDepositsAsync()
         {
-            SerializableResult<List<MaturedBlockDepositsModel>> matureBlockDeposits = 
-                await this.federationGatewayClient.GetMaturedBlockDepositsAsync(this.crossChainTransferStore.NextMatureDepositHeight, this.nodeLifetime.ApplicationStopping).ConfigureAwait(false);
+            // First ensure that we the node is out of IBD.
+            if (this.initialBlockDownloadState.IsInitialBlockDownload())
+            {
+                this.logger.Info("The CCTS will start processing deposits once the node is out of IBD.");
+                return true;
+            }
+
+            // Then ensure that the federation wallet is synced with the chain.
+            if (this.federationWalletManager.WalletTipHeight < this.chainIndexer.Tip.Height - FederationWalletTipSyncBuffer)
```

Comment:
Could an additional check be added to not run the CCTS sync when the wallet needs to be consolidated first?